paddle/fluid/operators/matmul_op.cc

```diff
@@ -324,8 +324,8 @@ class MatMulGradKernel : public framework::OpKernel<T> {
   }
 };
 
-framework::DDim GetDimForInput(const framework::InferShapeContext &ctx,
-                               std::string input_name) {
+static framework::DDim GetDimForInput(const framework::InferShapeContext &ctx,
+                                      std::string input_name) {
```

Comment:
```suggestion
                                      const std::string& input_name) {
```
That might not be performance-critical, but it still may save us some time

Reply:
AFAIK GetDimForInput is always called with const char*.
I think that the string construction happens in place here based on char ptr so there is no copying of string but passing a char*. If so, then it's not necessarily slower. I will use a char argument instead 
and create a string based on it inside the function explicitly. It will show better the intent behind it. I will also validate if it's exactly either 'X' or 'Y'.

Reply:
That sounds nice, thank you!